app/controllers/devise/registrations_controller.rb

```diff
@@ -97,6 +109,10 @@ def update_resource(resource, params)
     resource.update_with_password(params)
   end
 
+  def destroy_resource(resource, params)
+    resource.destroy_with_password(params)
```

Comment:
Here the `params` parameter is actually only the literal `current_password`, right? If that's the case, I'd either rename it or change it here to receive the full params hash and then pass only the `current_password` to `resource#destroy_with_password`